src/Arcus.Messaging.Pumps.ServiceBus.KeyRotation/ReAuthenticateMessageHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.BackgroundJobs.KeyVault.Events;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions.MessageHandling;
+using CloudNative.CloudEvents;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest.Azure;
+
+namespace Arcus.Messaging.Pumps.ServiceBus.KeyRotation
+{
+    /// <summary>
+    /// Represents an <see cref="IMessageHandler{TMessage, TMessageContext}"/> that processes <see cref="CloudEvent"/>s representing <see cref="SecretNewVersionCreated"/> events
+    /// that will eventually result in restarting an <see cref="AzureServiceBusMessagePump"/> instance.
+    /// </summary>
+    public class ReAuthenticateMessageHandler : IAzureServiceBusMessageHandler<CloudEvent>
```

Comment:
It's a long name, but maybe more explicit?
```suggestion
    public class MessagePumpAuthenticationRotatedMonitorMessageHandler : IAzureServiceBusMessageHandler<CloudEvent>
```

Reply:
😄 that's the burden of OOP , long names 😉 . We could rename it but let me think of something 🤔 .

Reply:
Renamed to `ReAuthenticateOnRotatedCredentialsMessageHandler`.